x-pack/plugins/apm/server/routes/profiling/get_service_host_names.ts

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { ProcessorEvent } from '@kbn/observability-plugin/common';
+import { kqlQuery, rangeQuery } from '@kbn/observability-plugin/server';
+import { HOST_HOSTNAME, SERVICE_NAME } from '../../../common/es_fields/apm';
+import { environmentQuery } from '../../../common/utils/environment_query';
+import { APMEventClient } from '../../lib/helpers/create_es_client/create_apm_event_client';
+
+export async function getServiceHostNames({
+  apmEventClient,
+  serviceName,
+  start,
+  end,
+  environment,
+  kuery,
+}: {
+  environment: string;
+  kuery: string;
+  serviceName: string;
+  start: number;
+  end: number;
+  apmEventClient: APMEventClient;
+}) {
+  const response = await apmEventClient.search('get_service_host_names', {
+    apm: {
+      events: [ProcessorEvent.metric],
```

Comment:
We've been trying to move out from processor events and using `apm.sources`, I'm not sure if you can use it in this case, what do you think @dgieselaar ? Not sure if should be transactionMetric or serviceTransactionMetric
```suggestion
      apm: {
        sources: [
          {
            documentType: ApmDocumentType.TransactionMetric,
            rollupInterval: RollupInterval.None,
          },
        ],
      },
```

Reply:
Thanks for that, I got it fixed here: https://github.com/elastic/kibana/pull/165360/commits/68a71dca4fc083bc1e4b18cd627983caaa8a716f